File: oder/src/main/java/com/lexuancong/oder/viewmodel/OrderVm.java
package com.lexuancong.oder.viewmodel;

import com.lexuancong.oder.model.Order;
import com.lexuancong.oder.model.OrderItem;
import com.lexuancong.oder.model.enum_status.DeliveryStatus;
import com.lexuancong.oder.model.enum_status.OrderStatus;

import java.math.BigDecimal;
import java.util.Set;

public record OrderVm(
        Long id,
        String email,
        ShippingAddressVm shippingAddressVm,
        String note,
        int numberItem,
        BigDecimal totalPrice,
        OrderStatus oderStatus,
        DeliveryStatus deliveryStatus,
        Set<OrderItemVm> orderItemVms
) {
    public static OrderVm fromModel(Order order , Set<OrderItem> orderItemSet){
        OrderVm orderVm = new OrderVm();
    }

}

File: nextjs-frontend_store/models/order/OrderVm.ts
import {OrderStatus} from "@/models/order/OrderStatus";
import dayjs, {Dayjs} from "dayjs";
import {DeliveryStatus} from "@/models/order/DeliveryStatus";
import {DeliveryMethod} from "@/models/order/DeliveryMethod";
import {OrderItemVm} from "@/models/order/OrderItemVm";

export interface OrderVm{
    id:number;
    orderStatus:OrderStatus;
    totalPrice: number;
    deliveryStatus: DeliveryStatus;
    deliveryMethod: DeliveryMethod;
    orderItemVms: OrderItemVm[];
    createdAt: Dayjs;

}